src/convertToObject.js

```diff
@@ -5,8 +5,21 @@
  *
  * @return {object}
  */
+
 function convertToObject(sourceString) {
-  // write your code here
+  const converted = sourceString
+    .split(';')
+    .map((el) => el.trim())
+    .filter((el) => el.length > 0)
+    .reduce((obj, curr) => {
+      const arrValue = curr.split(':').map((el) => el.trim());
```

Comment:
Consider adding error handling to manage cases where `curr.split(':')` does not result in an array with two elements. This can happen if the input string is not formatted correctly, leading to potential runtime errors.